src/Monolog/Handler/Handler.php

```diff
@@ -44,19 +44,29 @@ public function __destruct()
         }
     }
 
-    public function __sleep()
+    public function __serialize(): array
     {
         $this->close();
 
         $reflClass = new \ReflectionClass($this);
 
-        $keys = [];
+        $data = [];
         foreach ($reflClass->getProperties() as $reflProp) {
             if (!$reflProp->isStatic()) {
-                $keys[] = $reflProp->getName();
+                $data[$reflProp->getName()] = $reflProp->getValue($this);
             }
         }
 
-        return $keys;
+        return $data;
+    }
+
+    /**
+     * @param array<mixed> $data
+     */
+    public function __unserialize(array $data): void
```

Comment:
this method should be removed: the implementation is almost what PHP does natively (it does more, taking mangled names into account if any) - the implementation here actually prevents any implementation of `__wakeup` in a child class from working.